source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidRealTimeMultiplayerClient.cs

```diff
@@ -34,111 +56,689 @@ public AndroidRealTimeMultiplayerClient(AndroidJavaObject account)
+            {
+                AndroidTaskUtils.AddOnSuccessListener<AndroidJavaObject>(
+                    task,
+                    annotatedData =>
+                    {
+                        using (var invitationBuffer = annotatedData.Call<AndroidJavaObject>("get"))
+                        {
+                            int count = invitationBuffer.Call<int>("getCount");
+                            for (int i = 0; i < count; i++)
+                            {
+                                Invitation invitation;
+                                using (var invitationObject = invitationBuffer.Call<AndroidJavaObject>("get", i))
+                                {
+                                    invitation = AndroidJavaConverter.ToInvitation(invitationObject);
+                                }
+
+                                if (invitation.InvitationId == invitationId)
+                                {
+                                    callback(invitation);
+                                    return;
+                                }
+                            }
+
+                            OurUtils.Logger.e("Invitation with ID " + invitationId + " couldn't be found");
+                            fail(true);
+                        }
+                    });
+
+                AndroidTaskUtils.AddOnFailureListener(
+                    task,
+                    exception =>
+                    {
+                        OurUtils.Logger.e("Couldn't load invitations.");
+                        fail(true);
+                    });
+            }
+        }
+
+        private class RoomStatusUpdateCallbackProxy : AndroidJavaProxy
+        {
+            private RealTimeMultiplayerListener mListener;
+            private AndroidRealTimeMultiplayerClient mParent;
+
+            public RoomStatusUpdateCallbackProxy(AndroidRealTimeMultiplayerClient parent,
+                RealTimeMultiplayerListener listener) : base(
+                "com/google/games/bridge/RoomStatusUpdateCallbackProxy$Callback")
+            {
+                mListener = listener;
+                mParent = parent;
+            }
+
+            public void onRoomConnecting(AndroidJavaObject room)
+            {
+                mParent.mRoom = room;
+            }
+
+            public void onRoomAutoMatching(AndroidJavaObject room)
+            {
+                mParent.mRoom = room;
+            }
+
+            public void onPeerInvitedToRoom(AndroidJavaObject room, AndroidJavaObject participantIds)
+            {
+                handleParticipantStatusChanged(room, participantIds);
+            }
+
+            public void onPeerDeclined(AndroidJavaObject room, AndroidJavaObject participantIds)
+            {
+                handleParticipantStatusChanged(room, participantIds);
+            }
+
+            public void onPeerJoined(AndroidJavaObject room, AndroidJavaObject participantIds)
```

Comment:
Can you check how it will behave with a null object passed from java?

Reply:
Just ftr, when null is passed here from a java class, it doesn't give error until trying to call a method of this null instance (as expected). 

We now handle null room cases in RoomUpdateCallback and fail if it's null. But we leave null room case unhandled in RoomStatusUpdateCallback.